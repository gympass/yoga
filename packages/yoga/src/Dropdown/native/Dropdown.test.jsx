import React from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider, Dropdown } from '../..';

jest.useFakeTimers();

const dropdownProps = {
  label: 'Find an activity to love',
  options: [
    { label: 'Yoga', value: 'yoga' },
    { label: 'Crossfit', value: 'crossfit' },
    { label: 'Tenis', value: 'tenis' },
    { label: 'Soccer', value: 'soccer' },
    { label: 'Pilates', value: 'pilates' },
    { label: 'Run', value: 'running' },
  ],
};

describe('<Dropdown />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Dropdown {...dropdownProps} />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot when disabled', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Dropdown disabled {...dropdownProps} />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot when full', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Dropdown full {...dropdownProps} />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot when has a selected value', () => {
      const selectedOption = {
        label: 'Swimming',
        value: 'swimming',
        selected: true,
      };
      const props = dropdownProps;

      props.options.push(selectedOption);

      const { toJSON } = render(
        <ThemeProvider>
          <Dropdown disabled {...props} />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with error', () => {
      const { toJSON } = render(
        <ThemeProvider>
          <Dropdown {...dropdownProps} error="Please, select one activity" />
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
