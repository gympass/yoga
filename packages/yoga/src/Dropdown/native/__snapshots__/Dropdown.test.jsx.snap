// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> Snapshots should match snapshot 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    disabled={false}
    full={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D7D7E0",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "relative",
          "width": 320,
        },
      ]
    }
  >
    <Text
      disabled={false}
      inverted={false}
      light={false}
      size="medium"
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Object {
            "color": "#6B6B78",
          },
        ]
      }
    >
      Find an activity to love
    </Text>
    <RNSVGSvgView
      bbHeight={20}
      bbWidth={20}
      fill="#6B6B78"
      focusable={false}
      height={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4285229944,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`<Dropdown /> Snapshots should match snapshot when disabled 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    disabled={true}
    full={false}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#F5F5FA",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "relative",
          "width": 320,
        },
      ]
    }
  >
    <Text
      disabled={true}
      inverted={false}
      light={false}
      size="medium"
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Object {
            "color": "#D7D7E0",
          },
        ]
      }
    >
      Find an activity to love
    </Text>
    <RNSVGSvgView
      bbHeight={20}
      bbWidth={20}
      fill="#D7D7E0"
      focusable={false}
      height={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4292335584,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`<Dropdown /> Snapshots should match snapshot when full 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    disabled={false}
    full={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#D7D7E0",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "relative",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      disabled={false}
      inverted={false}
      light={false}
      size="medium"
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Object {
            "color": "#6B6B78",
          },
        ]
      }
    >
      Find an activity to love
    </Text>
    <RNSVGSvgView
      bbHeight={20}
      bbWidth={20}
      fill="#6B6B78"
      focusable={false}
      height={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4285229944,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`<Dropdown /> Snapshots should match snapshot when has a selected value 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    disabled={true}
    full={false}
    selected={
      Object {
        "label": "Swimming",
        "selected": true,
        "value": "swimming",
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#231B22",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "relative",
          "width": 320,
        },
      ]
    }
  >
    <Text
      disabled={true}
      inverted={false}
      light={false}
      selected={
        Object {
          "label": "Swimming",
          "selected": true,
          "value": "swimming",
        }
      }
      size="medium"
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Object {
            "color": "#231B22",
          },
        ]
      }
    >
      Swimming
    </Text>
    <RNSVGSvgView
      bbHeight={20}
      bbWidth={20}
      fill="#D7D7E0"
      focusable={false}
      height={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4292335584,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`<Dropdown /> Snapshots should match snapshot with error 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    disabled={false}
    error="Please, select one activity"
    full={false}
    selected={
      Object {
        "label": "Swimming",
        "selected": true,
        "value": "swimming",
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": "#231B22",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "position": "relative",
          "width": 320,
        },
      ]
    }
  >
    <Text
      disabled={false}
      inverted={false}
      light={false}
      selected={
        Object {
          "label": "Swimming",
          "selected": true,
          "value": "swimming",
        }
      }
      size="medium"
      style={
        Array [
          Object {
            "color": "#231B22",
            "fontFamily": "Rubik",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Object {
            "color": "#231B22",
          },
        ]
      }
    >
      Swimming
    </Text>
    <RNSVGSvgView
      bbHeight={20}
      bbWidth={20}
      fill="#231B22"
      focusable={false}
      height={20}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          Object {
            "opacity": 1,
          },
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      width={20}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4280490786,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGSvgView>
  </View>
</View>
`;
